htlcswitch/link_test.go

```diff
@@ -187,28 +187,30 @@ func TestChannelLinkRevThenSig(t *testing.T) {
 
 	const chanAmt = btcutil.SatoshiPerBitcoin * 5
 	const chanReserve = btcutil.SatoshiPerBitcoin * 1
-	aliceLink, bobChannel, batchTicker, start, restore, err :=
-		newSingleLinkTestHarness(t, chanAmt, chanReserve)
+	harness, err := newSingleLinkTestHarness(t, chanAmt, chanReserve)
 	require.NoError(t, err)
 
-	err = start()
+	err = harness.start()
 	require.NoError(t, err)
-	defer aliceLink.Stop()
+	defer harness.aliceLink.Stop()
 
-	alice := newPersistentLinkHarness(
-		t, aliceLink, batchTicker, restore,
+	alice := newPersistentLinkHarness(t, harness.aliceSwitch,
+		harness.aliceLink, harness.aliceBatchTicker,
+		harness.aliceRestore,
 	)
 
 	var (
-		coreLink  = aliceLink.(*channelLink)
+		//nolint: forcetypeassert
```

Comment:
Q: I wonder whether we miss some functions in the ChannelLink interface forcing us to type convert ?

Reply:
Definitely. I think our practice of breaking the glass on the interface in order to test is definitely a problem. It's just one I have a much longer story arc on solving. I'm aligned on this though. We shouldn't be doing it and I'm actively trying to dramatically clean up the htlcswitch package, but it's very delicate and there will be a lot of incremental steps towards getting us to a point where we can successfully deal with this.